### PR TITLE
feat: Kusto cluster system assigned managed identity

### PR DESCRIPTION
**Pull request by oZakari**
## Description
Add managed identity capability to cluster.

## Pipeline Reference

| Pipeline |
| -------- |
|    [avm.res.kusto.cluster](https://github.com/oZakari/bicep-registry-modules/actions/workflows/avm.res.kusto.cluster.yml)      |

## Type of Change

- [x] Update to CI Environment or utilities (Non-module affecting changes)
- [x] Azure Verified Module updates:
  - [ ] Bugfix containing backwards-compatible bug fixes, and I have NOT bumped the MAJOR or MINOR version in `version.json`:
    - [ ] Someone has opened a bug report issue, and I have included "Closes #{bug_report_issue_number}" in the PR description.
    - [ ] The bug was found by the module author, and no one has opened an issue to report it yet.
  - [x] Feature update backwards compatible feature updates, and I have bumped the MINOR version in `version.json`.
  - [ ] Breaking changes and I have bumped the MAJOR version in `version.json`.
  - [ ] Update to documentation

## Checklist

- [x] I'm sure there are no other open Pull Requests for the same update/change
- [x] I have run `Set-AVMModule` locally to generate the supporting module files.
- [ ] My corresponding pipelines / checks run clean and green without any errors or warnings

### CODE REVIEW

**Comment by oZakari** [comment]
@AlexanderSehr would you mind taking a glance at this error when you get a chance. I don't understand why it's looking for `systemAssignedMIPrincipalId` as the output: https://github.com/oZakari/bicep-registry-modules/actions/runs/10392382963/job/28777456223

**Comment by oZakari** [comment]
Hey @segraef, I believe I've correctly added an output for the principal ID, but the test is still saying it doesn't exist. Could you take a quick glance when you get a chance please?

File: avm/res/kusto/cluster/main.bicep [hunk 1]
```diff
@@ -370,6 +372,9 @@ output name string = kustoCluster.name
 @description('The location the resource was deployed into.')
 output location string = kustoCluster.location
 
+@description('The identity of the cluster.')
+output identity object = kustoCluster.identity
```
Comment (AlexanderSehr) [inline avm/res/kusto/cluster/main.bicep:376]: You can have this output if you want. But the one that is absolutely required is this one:
```bicep
@description('The principal ID of the system assigned identity.')
output systemAssignedPrincipalId string = kustoCluster.?identity.?principalId ?? ''
```

([spec reference](https://azure.github.io/Azure-Verified-Modules/specs/shared/interfaces/#managed-identities))

Comment (AlexanderSehr) [reply avm/res/kusto/cluster/main.bicep:376]: Hence the error message ;) 
image

If the error message did not help understand & solve the issue, please provide us with set feedback so we can work on that :) Somebody just wrote this text someday 😄

Comment (oZakari) [reply avm/res/kusto/cluster/main.bicep:379]: Thanks @AlexanderSehr, I updated the spec with the correct value in a PR here `systemAssignedMIPrincipalId` as well.

**Review by AlexanderSehr — APPROVED** [review]

**Comment by AlexanderSehr** [comment]
Great work @oZakari and sorry for the issue with the expected vs documented value.

**Pull request merged**